Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,505 @@
+            CHANTOR_DB = ['ChannelTorrents.channel_id', 'Torrent.torrent_id', 'infohash', '""', 'length',
+                          'category', 'status', 'num_seeders', 'num_leechers', 'ChannelTorrents.id',
+                          'ChannelTorrents.dispersy_id', 'ChannelTorrents.name', 'Torrent.name',
+                          'ChannelTorrents.description', 'ChannelTorrents.time_stamp', 'ChannelTorrents.inserted']
+
+            torrent_values = self.channelcast_db.getTorrentsFromChannelId(self.channel_id, True, CHANTOR_DB,
+                                                                          self.max_torrents)
+
+            listtor = self.torrent_mgr.create_torrents(torrent_values, True,
+                                                       {self.channel_id: self.channelcast_db.getChannel(
+                                                           self.channel_id)})
+
+            # dict {key_infohash(binary):Torrent(object-GUIDBTuple)}
+            self.unavail_torrent.update({t.infohash: t for t in listtor if t.infohash not in self.torrents})
+
+            # it's highly probable the checktor function is running at this time (if it's already running)
+            # if not running, start the checker
+
+            task_call = self.check_and_register_task(hexlify(self.source) + "_checktor", LoopingCall(self._check_tor))
+            if task_call:
+                self._logger.debug("Registering check torrent function")
+                task_call.start(self.check_torrent_interval, now=True)
+
+    def _on_database_updated(self, dummy_subject, dummy_change_type, dummy_infohash):
+        self.database_updated = True
+
+    def get_source_text(self):
+        return self.channel.name if self.channel else None
+
+
+class RSSFeedSource(BoostingSource):
+    """
+    Credit mining source from a RSS feed.
+
+    # supported list (tested) :
+    # http://bt.etree.org/rss/bt_etree_org.rdf
+    # http://www.mininova.org/rss.xml
+    # https://kat.cr (via torcache)
+
+    # not supported (till now)
+    # https://eztv.ag/ezrss.xml (https link)
+    """
+
+    def __init__(self, session, rss_feed, boost_settings, torrent_insert_cb):
+        BoostingSource.__init__(self, session, rss_feed, boost_settings, torrent_insert_cb)
+
+        self.feed_handle = None
+
+        self.torrent_store = self.session.lm.torrent_store
+        # keys : fake infohash, value : real infohash. Type : (length 20 string, binary)
+        self.fake_infohash_id = {}
+
+        self.title = ""
+        self.description = ""
+        self.total_torrents = 0
+
+        self.torrent_db = self.session.open_dbhandler(NTFY_TORRENTS)
+
+    def _load(self, rss_feed):
+        self.feed_handle = self.session.lm.ltmgr.get_session().add_feed(
+            {'url': rss_feed, 'auto_download': False, 'auto_map_handles': False})
+
+        def wait_for_feed(d=None):
+            """
+            Wait until the RSS feed is no longer updating.
```

Comment:
I've seen the code, but not like I wanted (especially with the needs of channel parameter there). Since I only need the url, I think I just use `getPage` and `feedparser`, and parse it manually here.